subway_structure.py

```diff
@@ -1740,7 +1743,7 @@ def extract_routes(self):
                     ):
                         continue
 
-                route = Route(el, self, master)
+                route = self.__class__.route_class(el, self, master)
```

Comment:
нет необходимости обращаться к __class__ в обычном методе класса (только если метод статичный и нужно значение атрибута класса, но тогда возникнет вопрос, почему он сделан статичным, если ему нужен атрибут класса)
```suggestion
                route = self.route_class(el, self, master)
```

Reply:
Мне нужно реализовать полиморфное порождающее поведение. Т.е. будут классы class CityEx(City) и RouteEx(Route), с тем чтобы CityEx внутри себя создавал экземпляры RouteEx. Я применяю идею паттерна проектирования "Фабричный метод", просто вместо метода использую классовую переменную. Стоит ли создавать методы вида `def get_route_class(self): return Route`, с инстанцированием в виде `route = self.get_route_class()(args)`, только чтобы избавиться от обращения к атрибуту \_\_class\_\_? Разве такое обращение порицается? Может, тут подойдёт другой паттерн?

Reply:
\+